test/test_heuristic.cpp

```diff
@@ -81,6 +81,84 @@ TEST(Functionality, NodeCostCalculation) {
               tolerance);
 }
 
+TEST(Functionality, HeuristicBenchmark) {
+  const double tolerance = 1e-10;
+  /*
+      3
+     / \
+    4   2
+    |   |
+    0---1
+  */
+  Architecture      architecture{};
+  const CouplingMap cm = {{0, 1}, {1, 0}, {1, 2}, {2, 1}, {2, 3},
+                          {3, 2}, {3, 4}, {4, 3}, {4, 0}, {0, 4}};
+  architecture.loadCouplingMap(5, cm);
+
+  qc::QuantumComputation qc{5, 5};
+  qc.x(2, qc::Control{4});
+  qc.x(1, qc::Control{3});
+  qc.x(1, qc::Control{4});
+
+  qc.barrier({0, 1, 2, 3, 4});
+  for (size_t i = 0; i < 5; ++i) {
+    qc.measure(static_cast<qc::Qubit>(i), i);
+  }
```

Comment:
Just a side note: at some point, we should really add a `measure_all` convenience function to the `QuantumComputation` class in the QFR similar to what Qiskit offers.

Reply:
Yes, definitely. Especially since the mapper will initially not map any qubits that are not used nor measured (which makes sense for practical settings, but is often inconvenient during testing) So measuring all qubits is a nice work-around to force the mapper to map all qubits in the beginning according to the policy `Configuration::initialLayout`.

Reply:
Right. I'll put it on my list of todos 👍🏻